packages/smooth_app/lib/pages/scan/camera_image_cropper.dart

```diff
@@ -125,3 +198,72 @@ class CameraImageCropper extends AbstractCameraImageGetter {
     return planeData;
   }
 }
+
+/// On iOS, coordinates are hardcoded in the native code (0, 0)
+/// [https://github.com/danjodanjo/google_ml_barcode_scanner/blob/master/ios/Classes/MLKVisionImage%2BFlutterPlugin.m#L136]
+///
+/// The only crop we can do is to shrink the width and/or the height
+class _CameraImageCropperImplIOS extends CameraImageCropper {
+  _CameraImageCropperImplIOS(
+    final CameraImage cameraImage,
+    final CameraDescription cameraDescription, {
+    required double width01,
+    required double height01,
+  })  : assert(width01 > 0 && width01 <= 1),
+        assert(height01 > 0 && height01 <= 1),
+        super._(
+          cameraImage,
+          cameraDescription,
+          width01: width01,
+          height01: height01,
+        ) {
+    _computeCropParameters();
+  }
+
+  late int _width;
+  late int _height;
+
+  void _computeCropParameters() {
+    if (orientation == 0) {
+      _width = _computeWidth();
+      _height = _computeHeight();
+    } else if (orientation == 90) {
+      _width = _computeWidth();
+      _height = _computeHeight();
+    } else {
+      throw Exception('Orientation $orientation not dealt with for the moment');
+    }
+  }
+
+  @override
+  Size getSize() => Size(
+        _width.toDouble(),
+        _height.toDouble(),
+      );
+
+  // Same implementation as [CameraImageFullGetter]
```

Comment:
I'm a bit surprised: what's the added value if you copy the whole set of planes?
We would be better off with `CameraImageFullGetter`, wouldn't we?

Reply:
Basically it's a CameraImageFullGetter with a custom size.
Since this class still an "ImageCropper", using a mixin may be a solution

Reply:
> Basically it's a CameraImageFullGetter with a custom size.

That's what I read, but I have serious doubts about copying a full size while saying it's half size. It's not worth the complication anyway as there's no performance gain.

> Since this class still an "ImageCropper", using a mixin may be a solution

I'm not familiar with `mixin`, but from what I read that's too complex for what we want. Assuming your idea of "full copy but half `getSize`" is correct, you could just `extend` `CameraImageFullGetter` and override `getSize`.
In the end we don't need a `CameraImageCropper` anyway, we need an `AbstractCameraImageGetter`.